cmd/aws-actuator/main.go

```diff
@@ -435,221 +302,92 @@ func bootstrapCommand() *cobra.Command {
+				return err
 			}
 
-			err = wait.Poll(pollInterval, timeoutPoolAWSInterval, func() (bool, error) {
-				glog.Info("Deploying cluster-api controllers")
-				if output, err := cmdRun("kubectl", "--kubeconfig=kubeconfig", "apply", fmt.Sprintf("-f=%v", path.Join(manifestsDir, "provider-components.yml"))); err != nil {
-					glog.Infof("Unable to apply %v manifest: %v\n%v", path.Join(manifestsDir, "provider-components.yml"), err, string(output))
-					return false, nil
+			clusterFramework.ErrNotExpected = func(err error) {
+				if err != nil {
+					panic(err)
 				}
-				return true, nil
-			})
-			if err != nil {
-				glog.Errorf("Unable to deploy cluster-api controllers: %v", err)
 			}
 
-			testCluster := &clusterv1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "tb-asg-35",
-					Namespace: "test",
-				},
-				Spec: clusterv1.ClusterSpec{
-					ClusterNetwork: clusterv1.ClusterNetworkingConfig{
-						Services: clusterv1.NetworkRanges{
-							CIDRBlocks: []string{"10.0.0.1/24"},
-						},
-						Pods: clusterv1.NetworkRanges{
-							CIDRBlocks: []string{"10.0.0.1/24"},
-						},
-						ServiceDomain: "example.com",
-					},
-				},
+			clusterFramework.By = func(msg string) {
+				glog.Info(msg)
 			}
 
-			err = wait.Poll(pollInterval, timeoutPoolAWSInterval, func() (bool, error) {
-				glog.Infof("Deploying cluster resource")
-
-				if err := createCluster(tc, testCluster); err != nil {
-					glog.Infof("Unable to deploy cluster manifest: %v", err)
-					return false, nil
-				}
-
-				return true, nil
-			})
-			if err != nil {
-				glog.Errorf("Unable to deploy cluster resource %v", err)
-			}
-
-			glog.Infof("Reading worker user data manifest from %v", path.Join(manifestsDir, "worker-userdata.yaml"))
-			workerUserDataSecret, err := readSecretManifest(path.Join(manifestsDir, "worker-userdata.yaml"))
+			glog.Info("Waiting for all nodes to come up")
+			err = clusterFramework.WaitForNodesToGetReady(1)
 			if err != nil {
 				return err
 			}
 
-			glog.Infof("Generating worker machine set user data for master listening at %v", masterMachinePrivateIP)
-			workerUserDataSecret, err = generateWorkerUserData(masterMachinePrivateIP, workerUserDataSecret)
-			if err != nil {
-				return fmt.Errorf("unable to generate worker user data: %v", err)
-			}
-
-			if err := createSecret(tc, workerUserDataSecret); err != nil {
+			glog.Infof("Creating %q namespace", testNamespace.Name)
+			if _, err := clusterFramework.KubeClient.CoreV1().Namespaces().Create(testNamespace); err != nil {
 				return err
 			}
 
-			glog.Infof("Reading worker machine manifest from %v", path.Join(manifestsDir, "worker-machineset.yaml"))
-			workerMachineSet, err := readMachineSetManifest(
-				&manifestParams{
-					ClusterID: machinePrefix,
-				},
-				path.Join(manifestsDir, "worker-machineset.yaml"),
-			)
+			clusterFramework.DeployClusterAPIStack(testNamespace.Name, "openshift/origin-aws-machine-controllers:v4.0.0", "")
```

Comment:
Do we always want 4.0.0 here?

Reply:
Currently yes. Later, we can parametrize the image.